file-formats/aifa/type/zif_aff_aifa_v1.intf.abap

```diff
@@ -0,0 +1,103 @@
+INTERFACE zif_aff_aifa_v1
+   PUBLIC.
+
+  "! $values { @link zif_aff_aifa_v1.data:co_error_handling }
+  "! $default { @link zif_aff_aifa_v1.data:co_error_handling.ignore_data }
+  TYPES ty_error_handling_type TYPE c LENGTH 1.
+  CONSTANTS:
+    BEGIN OF co_error_handling,
+      "! <p class="shorttext">Treat As Error</p>
+      "! Treat as error if check is not successful
+      trigger_error TYPE c LENGTH 1 VALUE '',
+      "! <p class="shorttext">Ignore Data</p>
+      "! Ignore data if check is not successful
+      ignore_data   TYPE c LENGTH 1 VALUE 'I',
+    END OF co_error_handling.
+
+  TYPES:
+    "! <p class="shorttext">Field To Check</p>
+    "! Field to check
+    BEGIN OF ty_field_to_check,
+      "! <p class="shorttext">Field Name</p>
+      "! Field name
+      name TYPE string,
+    END OF ty_field_to_check,
+    "! <p class="shorttext">Fields To Check</p>
+    "! Fields to check
+    ty_fields_to_check TYPE STANDARD TABLE OF ty_field_to_check WITH DEFAULT KEY.
+
+  TYPES:
+    "! <p class="shorttext">General Information</p>
+    "! General information
+    BEGIN OF ty_general_information,
+      "! <p class="shorttext">Namespace</p>
+      "! Namespace
+      "! $required
+      namespace           TYPE c LENGTH 15,
+      "! <p class="shorttext">AIF Action</p>
+      "! AIF action
+      "! $required
+      aif_action          TYPE c LENGTH 20,
+      "! <p class="shorttext">Main Component Type</p>
+      "! Main component type
+      main_component_type TYPE zif_aff_types_v1=>ty_object_name_30,
+      "! <p class="shorttext">Class</p>
+      "! Class
+      class               TYPE zif_aff_types_v1=>ty_object_name_30,
```

Comment:
I guess we are talking about an ABAP class. What is the purpose of the class? Would it make sense to add the purpose to the field name?

Reply:
I prefer some addition like it is done here:
https://github.com/SAP/abap-file-formats/blob/158e208851c97f795e85738b43fc1e761b18c626/file-formats/chko/examples/z_aff_example_chko.chko.json#L8

However, I have just checked the repository. For AIFP, also "class" was used. So this is kind of consistent at least in your domain:
https://github.com/SAP/abap-file-formats/blob/158e208851c97f795e85738b43fc1e761b18c626/file-formats/aifp/aifp-v1.json#L377

Reply:
I reopened this discussion since the issue also came up during UX review. We suggest to use 
```suggestion
      "! <p class="shorttext">Implementing Class</p>
      "! Implementing Class
      implementing_class               TYPE zif_aff_types_v1=>ty_object_name_30,
```
In my opinion, this is different to AIFP here, since there, `class` is part of the component `Custom Implementation`. Here, it is in `General Information`. So in my opinion, the purpose of the class would be helpful.